lib/connectors/common/coder.js

```diff
@@ -36,11 +36,18 @@ module.exports.decode = function decode(msg) {
   }
 
   // decode use protobuf
+  // if gzip is enabled, it still tests protobuf first since it's a specific compression method.
   if (this.protobuf && this.protobuf.getProtos().client[route]) {
     msg.body = this.protobuf.decode(route, msg.body);
   } else if (this.decodeIO_protobuf &&
              this.decodeIO_protobuf.check(Constants.RESERVED.CLIENT, route)) {
     msg.body = this.decodeIO_protobuf.decode(route, msg.body);
+  } else if (this.gzip) {
+    try {
+      msg.body = this.gzip.decode(route, msg.body);
+    } catch (ex) {
+      msg.body = {};
```

Comment:
Is it worth having an error log here?

Reply:
I followed the same format as other decoders, and I don't find error logs useful here, and it can make us vulnarable to a DoS attack.
My thoery is that if this decoder starts to fail, it is probably related to a wrong implementation (that we should be able to catch during developing), or an attack.

Reply:
> It can make us vulnerable to a DoS attack

It shouldn't be an approach to hide error logs to mitigate a DoS attack (even because we're talking about the logging and not necessarily a behavior change in the application). In this case, we should rely on the upper layers to filter that.

> I followed the same format as other decoders

We can follow that. (=